### PR TITLE
Added "log.origin.function" to caller

### PR DESCRIPTION
**Pull request by vispariana**
This adds the ECS standard "log.origin.function" field to the caller fields, alongside "log.origin.file.line" and "log.origin.file.name".

### CODE REVIEW

File: encoder.go [hunk 1]
```diff
@@ -65,6 +66,7 @@ func (e *CallerEncoder) UnmarshalText(text []byte) error {
 func encodeCaller(c *caller, enc zapcore.PrimitiveArrayEncoder) {
 	// this function can only be called internally so we have full control over it
 	// and can ensure that enc is always of type zapcore.ArrayEncoder
+	c.Function = runtime.FuncForPC(c.PC).Name()
```
Comment (axw) [inline encoder.go:69]: I don't think the encoder should be responsible for determining the function name, only for encoding it if it is recorded.

Zap will already record Function if `WithCaller(true)` is used: https://github.com/uber-go/zap/blob/a55bdc32f526699c3b4cc51a2cc97e944d02fbbf/logger.go#L374-L382

**Review by axw — COMMENTED** [review]
Thanks for the PR, @vispariana! I don't think it's the encoder's responsibility to be determining the function name. Otherwise looks good.

File: encoder.go (hunk 1 above)
Comment (vispariana) [reply encoder.go]: You're right. Thanks for reviewing. Encoder tests manually marshal the EntryCaller, and EntryCaller's Function fields gets filled with zap's unexported check() method, so the function field was empty in tests. I fixed the tests.

**Review by axw — APPROVED** [review]
Perfect! Thanks for your contribution and for making the update.

**Pull request merged**